lib/concurrent/executor/ruby_thread_pool_executor.rb

```diff
@@ -101,6 +101,11 @@ def worker_died(worker)
       synchronize { ns_worker_died worker }
     end
 
+    # @!visibility private
+    def worker_task_completed
+      @completed_task_count += 1
```

Comment:
This callback needs synchronized. Can you please look to `#worker_died` as an example?